src/language-server/project/rover/project.ts

```diff
@@ -108,30 +162,40 @@ export class RoverProject extends GraphQLProject {
       console.error({ err });
     });
 
+    connection.onNotification(
+      PublishDiagnosticsNotification.type,
+      this.documents.handlePartDiagnostics.bind(this.documents),
```

Comment:
Trying hard to get into the habit of using `.bind` instead of arrow functions where possible to prevent the risk of closure-related memory leaks 🙈